src/modules/getBullsAndCows.js

```
@@ -13,7 +13,38 @@
  * Example: { bulls: 1, cows: 2 }
  */
 function getBullsAndCows(userInput, numberToGuess) {
-  /* Write your code here */
+  const user = String(userInput).split('');
+  const correct = String(numberToGuess).split('');
+
+  let bulls = 0;
+  let cows = 0;
+
+  const correctUsed = [false, false, false, false];
+  const userUsed = [false, false, false, false];
+
+  for (let i = 0; i < 4; i++) {
+    if (user[i] === correct[i]) {
+      bulls++;
+      correctUsed[i] = true;
+      userUsed[i] = true;
+    }
+  }
+
+  for (let i = 0; i < 4; i++) {
+    if (userUsed[i]) {
+      continue;
+    }
+
+    for (let j = 0; j < 4; j++) {
+      if (!correctUsed[j] && user[i] === correct[j]) {
+        cows++;
+        correctUsed[j] = true;
+        break;
+      }
+    }
+  }
+
+  return { bulls, cows };
```

Returning `{ bulls, cows }` is correct. Given the marking logic above, the sum of `bulls + cows` cannot exceed 4. If you add defensive length checks (recommended), consider what the function should return/throw for invalid lengths and document that behavior in JSDoc.